framework/src/problems/FEProblemBase.C

```diff
@@ -8177,6 +8152,35 @@ FEProblemBase::notifyWhenMeshChanges(MeshChangedInterface * mci)
   _notify_when_mesh_changes.push_back(mci);
 }
 
+void
+FEProblemBase::notifyWhenMeshDisplaces(MeshDisplacedInterface * mdi)
+{
+  _notify_when_mesh_displaces.push_back(mdi);
+}
+
+bool
+FEProblemBase::updateDisplacement()
+{
+  if (!_displaced_problem)
+    return false;
+
+  if (getCurrentExecuteOnFlag() == EXEC_LINEAR || getCurrentExecuteOnFlag() == EXEC_NONLINEAR)
+    computeSystems(EXEC_PRE_DISPLACE);
```

Comment:
I think I'd rather a boolean parameter than guessing based on the exec flag

Reply:
You want an argument like `FEProblemBase::execute`?

Reply:
Why not a Boolean for computeSystems instead?